### PR TITLE
Add data fetcher to fetch data from degra

### PR DESCRIPTION
**Pull request by wvffle**
When it comes to `tabela_rozklad` tags:
- `<DZIEN>` is a day from `1` to `7`
- `<ILOSC>` is a number of intervals
- `<GODZ>` is an id of interval
- `<TYG>` decides whether or not the classes are in odd (`1`), even (`2`) or both weeks (`0`) 
  - Could be implemented as binary flags:
    ```js
    const flags = 3 - week
    const isEven = flags & 0b01
    const isOdd = flags & 0b10
    const isBoth = isOdd && isEven
    ```
- `<RODZ>` is a type of classes
- the rest are ids
image

### CODE REVIEW

**Comment by wvffle** [comment]
@all-contributors please add @wvffle for infrastructure, maintenance and code

**Comment by wvffle** [comment]
@all-contributors please add @Sylwia0506 for maintenance and code

**Review by Sylwia0506 — DISMISSED** [review]

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Subject.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    public int getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getShort_name() {
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Subject.java]: getShortName

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    public String getType() {
+        return type;
+    }
+
+    public int getGroup() {
+        return group;
+    }
+
+    public int getId_studies() {
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: getStudiesId

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+
+public class Schedule extends ScheduleEntry {
+    public static final String TAG_NAME = "tabela_rozklad";
+    private final int id;
+    private final int day;
+    private final int hour;
+    private final int amount;
+    private final int week;
+    private final int id_teacher;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: teacherId

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    private final int id_room;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: roomId

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    private final int id_subject;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: subjectId

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    private final String type;
+    private final int group;
+    private final int id_studies;
+    private final int term;
+    private final int id_speciality;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: specialityId

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    private final int day;
+    private final int hour;
+    private final int amount;
+    private final int week;
+    private final int id_teacher;
+    private final int id_room;
+    private final int id_subject;
+    private final String type;
+    private final int group;
+    private final int id_studies;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: degreeId

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 8]
```diff
@@ -0,0 +1,94 @@
+package net.wvffle.android.pb.schedule.api.db;
+
+
+public class Schedule extends ScheduleEntry {
+    public static final String TAG_NAME = "tabela_rozklad";
+    private final int id;
+    private final int day;
+    private final int hour;
+    private final int amount;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: intervalCount

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    private final int week;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: weekFlags

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 10]
```diff
@@ -0,0 +1,94 @@
+package net.wvffle.android.pb.schedule.api.db;
+
+
+public class Schedule extends ScheduleEntry {
+    public static final String TAG_NAME = "tabela_rozklad";
+    private final int id;
+    private final int day;
+    private final int hour;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: intervalId

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Teacher.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
+package net.wvffle.android.pb.schedule.api.db;
+
+
+public class Teacher extends ScheduleEntry {
+    public static final String TAG_NAME = "tabela_nauczyciele";
+    private final int id;
+    private final String surname;
+    private final String first_name;
+    private final String first_initial;
+    private final int id_title;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Teacher.java]: titleId

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Teacher.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
+package net.wvffle.android.pb.schedule.api.db;
+
+
+public class Teacher extends ScheduleEntry {
+    public static final String TAG_NAME = "tabela_nauczyciele";
+    private final int id;
+    private final String surname;
+    private final String first_name;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Teacher.java]: firstName

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Teacher.java [hunk 13]
```diff
@@ -0,0 +1,45 @@
+package net.wvffle.android.pb.schedule.api.db;
+
+
+public class Teacher extends ScheduleEntry {
+    public static final String TAG_NAME = "tabela_nauczyciele";
+    private final int id;
+    private final String surname;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Teacher.java]: lastName

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Teacher.java [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    private final String first_name;
+    private final String first_initial;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Teacher.java]: initials

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Subject.java [hunk 15]
```diff
@@ -0,0 +1,33 @@
+package net.wvffle.android.pb.schedule.api.db;
+
+
+public class Subject extends ScheduleEntry {
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Subject.java:4]: Class

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+    private final int amount;
+    private final int week;
+    private final int id_teacher;
+    private final int id_room;
+    private final int id_subject;
+    private final String type;
+    private final int group;
+    private final int id_studies;
+    private final int term;
```
Comment (wvffle) [inline app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: semester

File: app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java (hunk 9 above)
Comment (wvffle) [reply app/src/main/java/net/wvffle/android/pb/schedule/api/db/Schedule.java]: do tego tez trzeba bedzie stworzyc oddzielna klase z flagami, cos w stylu:
```java
class WeekFlags {
  public static final int EVEN = 1;
  public static final int ODD = 2;
}
```

**Review by Sylwia0506 — APPROVED** [review]

**Pull request merged**